adapter-kafka/src/main/java/io/nosqlbench/adapter/kafka/ops/OpTimeTrackKafkaConsumer.java

```diff
@@ -191,6 +209,16 @@ private void updateE2ELatencyMetric(ConsumerRecord<String, String> record) {
         }
     }
 
+    private void checkAndUpdateMessageErrorCounter(ConsumerRecord<String, String> record) {
+        String msgSeqIdStr = new String(record.headers().lastHeader(PulsarAdapterUtil.MSG_SEQUENCE_NUMBER).value());
```

Comment:
what happens if there is no such header ?
do we see here a null or does it throw an exception ?

Reply:
In case there is not such a key among the header keys, a NPE would be thrown, as `value()` is applied on `null`.

Reply:
so we should handle that case (null) the same way as a blank string

Reply:
Now, I skip the metrics generation in case it's null and warn the user about it.